src/averageAges.js

```diff
@@ -55,7 +69,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredPeople = people.filter(person => (
+    people.find(mother => mother.name === person.mother) && (onlyWithSon
+      ? person.sex === 'm'
+      : true)
+  ));
+
+  const diffArr = filteredPeople.map(person =>
```

Comment:
in this case, it's clear what is it, but it's better not to use short forms like `diffArr`